### PR TITLE
CRI: Add a call to reopen log file for a container

### LINKED ISSUE #58823: CRI: Implement container log rotation
TL;DR: We want to allow kubelet to rotate the container stdout/stderr logs, and add a CRI call to ask the runtime to reopen the log file.

kubelet decides the log directory structure and passes to the runtime the path where container log should be stored. Given the level of knowledge and control kubelet has, it is currently the best candidate to rotate the logs (vs. runtime or a separate daemon). The detailed evaluation is in the [proposal doc](https://docs.google.com/document/d/1oQe8dFiLln7cGyrRdholMsgogliOtpAzq6-K3068Ncg/edit?usp=sharing).

IMO, the initial version should target a simple policy of size limit per file. We can iterate and add more advanced features based on disk management in the future. The stretch goal for this release will be supporting reading logs (`kubectl logs`) from rotated log files.

As for the docker integration (i.e., `--container-runtime=docker`) that does not fully implement CRI logging, kubelet should skip log rotation.

/cc @kubernetes/sig-node-feature-requests @dchen1107 @Random-Liu @mrunalp @resouer @feiskyer

### PR DESCRIPTION
**Pull request by yujuhong**
This allows a daemon external to the container runtime to rotate the log
file, and then ask the runtime to reopen the files.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #58823

**Release note**:
```release-note
CRI: Add a call to reopen log file for a container. 
```

### CODE REVIEW

**Comment by mrunalp** [comment]
:+1:

**Review by resouer — COMMENTED** [review]
LGTM from KataContainers view, thanks!

File: pkg/kubelet/apis/cri/v1alpha1/runtime/api.proto [hunk 1]
```diff
@@ -1153,3 +1157,11 @@ message MemoryUsage {
     // The amount of working set memory in bytes.
     UInt64Value working_set_bytes = 2;
 }
+
+message ReopenContainerLogRequest {
+    // ID of the container for which to retrieve status.
```
Comment (feiskyer) [inline pkg/kubelet/apis/cri/v1alpha1/runtime/api.proto]: typo: s/to retrieve status/to reopen container log file/.

Comment (yujuhong) [reply pkg/kubelet/apis/cri/v1alpha1/runtime/api.proto]: Done.

**Comment by yujuhong** [comment]
Fixed the testing fixtures in the commit. Reapplying lgtm.

**Pull request merged**